core/profiles/standard/standard.install

```diff
@@ -550,6 +556,7 @@ function standard_install() {
     'view files',
     $filtered_html_permission,
   );
+  $editor_permissions = array_merge($editor_permissions,  $file_type_permissions);
```

Comment:
This approach (creating a new array and then merging) is slightly inefficient. I suggest the following (pasting the whole chunk for clarity):

```php
// Create a default editor role, for users who need to be able to edit
  // content, but do not need to access advanced admin features.
  $editor_role = new stdClass();
  $editor_role->name = 'editor';
  $editor_role->label = 'Editor';
  $editor_role->description = 'Can edit most content on the site, but cannot modify site configuration settings.';
  $editor_role->weight = 2;
  user_role_save($editor_role);

  $editor_permissions = array(
    'view own unpublished content',
    'view any unpublished content',
    'view revisions',
    'revert revisions',
    'delete revisions',
    'create page content',
    'edit own page content',
    'edit any page content',
    'delete own page content',
    'create post content',
    'edit own post content',
    'edit any post content',
    'delete own post content',
    'create card content',
    'edit own card content',
    'edit any card content',
    'delete own card content',
    'view hidden paths',
    'access dashboard',
    'view the administration theme',
    'access administration bar',
    'access content overview',
    'access administration pages',
    'access site in maintenance mode',
    'flush caches',
    'edit terms in tags',
    'delete terms in tags',
    'administer taxonomy',
    'administer nodes',
    'search content',
    'use advanced search',
    $filtered_html_permission,
  );

  $file_types = file_permissions_get_configured_types();
  foreach ($file_types as $file_type) {
    $editor_permissions[] = "edit own $file_type files";
    $editor_permissions[] = "download own $file_type files";
  }
  user_role_grant_permissions('editor', $editor_permissions);
```

Reply:
Basically I've moved the loop AFTER the rest of the permissions have been added to the array.

Reply:
This makes sense. Testing now.